src/python/pants/backend/native/tasks/c_compile.py

```diff
@@ -29,16 +28,8 @@ def implementation_version(cls):
   def subsystem_dependencies(cls):
     return super(CCompile, cls).subsystem_dependencies() + (CCompileSettings.scoped(cls),)
 
-  @memoized_property
-  def _c_compile_settings(self):
-    return CCompileSettings.scoped_instance(self)
-
-  @memoized_property
-  def _c_toolchain(self):
-    return self._request_single(LLVMCToolchain, self._native_toolchain).c_toolchain
-
   def get_compile_settings(self):
```

Comment:
Just checking you mean to drop the memoization when inlining here and in cpp_compile.py below.

Reply:
This bit was intentional! This was in response to clearing up the docstrings for these abstractmethods in NativeTask to make it clear that this method is only called once (see [NativeTask](https://github.com/pantsbuild/pants/blob/c15cacfb7c94ef14628a5d295af93c21ef7f8691/src/python/pants/backend/native/tasks/native_compile.py#L138)). If it would seem to make more sense to let the leaf tasks handle the memoization, we can move it there.